### PR TITLE
SWIP-883 Phone number format validation logic and tests

### PR DESCRIPTION
**Pull request by cgtabacaru**
Finalising [SWIP-883](https://dfedigital.atlassian.net.mcas.ms/browse/SWIP-883) by adding the regex UK phone number format validation and unit tests for the validator.

### CODE REVIEW

**Review by SamNich-Hippo — APPROVED** [review]

File: apps/user-management/apps/frontend.Test/UnitTests/Validators/AccountDetailsValidatorTests.cs [hunk 1]
```diff
@@ -160,4 +160,47 @@ public void WhenAnyInputNotSupplied_WhenRequiresPhoneNumber_HasValidationErrors(
...
+    [Theory]
+    [InlineData("01253 123 123")]
+    [InlineData("01253123123")]
+    [InlineData("01253 12 31 23")]
+    [InlineData("+447650123123")]
+    [InlineData("+447650 123 123")]
+    [InlineData("+447650 12 31 23")]
+    [InlineData("07542123543")]
+    [InlineData("07542 123 543")]
+    [InlineData("07542 12 35 43")]
```
Comment (MatthewWilkes) [inline apps/user-management/apps/frontend.Test/UnitTests/Validators/AccountDetailsValidatorTests.cs:193]: Landlines won't work with the +44 prefix here, such as `+44 20 7123 4567`, despite it being valid.
Is the requirement for it to be a UK number from the client? What about people moving from abroad? Or residents of Ireland?

File: apps/user-management/apps/frontend/Validation/Common/CommonValidators.cs [hunk 2]
```diff
@@ -31,18 +31,16 @@ public static void PhoneNumberValidation<T>(this IRuleBuilder<T, string?> ruleBu
...
             .Must(phoneNumber =>
-                phoneNumber is not null && new[] { internationalMobileRegex, mobileRegex, landlineRegex }
-                    .Any(pattern => Regex.IsMatch(phoneNumber, pattern))
-            )
+            {
+                if (string.IsNullOrWhiteSpace(phoneNumber))
+                    return false;
+
+                var phoneNumberNoSpaces = Regex.Replace(phoneNumber, @"\s+", "");
+                return Regex.IsMatch(phoneNumberNoSpaces, @"^(?:\+447\d{9}|07\d{9}|0[12]\d{8,9})$");
```
Comment (MatthewWilkes) [inline apps/user-management/apps/frontend/Validation/Common/CommonValidators.cs:42]: Have you considered https://www.nuget.org/packages/libphonenumber-csharp ? Google's libphonenumber is great for validating the weirdness of phone numbers.

File: apps/user-management/apps/frontend.Test/UnitTests/Validators/AccountDetailsValidatorTests.cs (hunk 1 above)
Comment (cgtabacaru) [reply apps/user-management/apps/frontend.Test/UnitTests/Validators/AccountDetailsValidatorTests.cs:193]: Thanks, Matt. I know they don't, but went by error message for the core formats for the time being. We've gone with UK numbers, but I will share you question with Victoria. I will also check with Victoria about country code prefixed landlines and amend as needed.

File: apps/user-management/apps/frontend/Validation/Common/CommonValidators.cs (hunk 2 above)
Comment (cgtabacaru) [reply apps/user-management/apps/frontend/Validation/Common/CommonValidators.cs:42]: Thanks, Matt. I have, but it felt like overkill given our requirements for UK phone numbers. Also the fact that phone number will be captured for the initial/primary coordinators only, so around 150 to start with, with some changes over the years as staff leaves. Statistically, I figured there will be very few, if any edge cases. Happy to revisit the solution though if we feel this doens't cut it.

File: apps/user-management/apps/frontend.Test/UnitTests/Validators/AccountDetailsValidatorTests.cs (hunk 1 above)
Comment (cgtabacaru) [reply apps/user-management/apps/frontend.Test/UnitTests/Validators/AccountDetailsValidatorTests.cs:193]: @MatthewWilkes I've checked with Victoria and confirmed the requirement for UK phone numbers. I'll add the prefixed landline in as a supported format, so thank you for raising that. 🙌
